components/tidb_query_common/src/metrics.rs

```diff
@@ -29,16 +30,45 @@ make_auto_flush_static_metric! {
     }
 }
 
-lazy_static::lazy_static! {
+make_static_metric! {
+    pub label_enum MemTrace {
+        aggr_fast_hash,
+        aggr_slow_hash,
+        aggr_stream,
+        aggr_simple,
+        top_n,
+        index_scan,
+        table_scan,
+        limit,
+        projection,
+        selection,
+        chunk,
+    }
+
+    pub struct MemTraceGauge : IntGauge {
+        "type" => MemTrace,
+    }
+}
+
+lazy_static! {
+    pub static ref MEMTRACE_QUERY_EXECUTOR: MemTraceGauge = register_static_int_gauge_vec!(
+        MemTraceGauge,
+        "tikv_query_executor_memory_usage",
+        "Coprocessor query executor memory usage",
+        &["type"]
+    )
+    .unwrap();
     static ref COPR_EXECUTOR_COUNT: IntCounterVec = register_int_counter_vec!(
         "tikv_coprocessor_executor_count",
         "Total number of each executor",
         &["type"]
     )
     .unwrap();
-}
-
-lazy_static::lazy_static! {
     pub static ref EXECUTOR_COUNT_METRICS: LocalCoprExecutorCount =
         auto_flush_from!(COPR_EXECUTOR_COUNT, LocalCoprExecutorCount);
 }
+
+pub trait MemoryTrace {
```

Comment:
It's weird to define `MemoryTrace` trait in `metrics.rs`

Reply:
Sure, where should it go and for future reference, why do you think it's weird?